service/matching/task_queue_partition_manager.go

```diff
@@ -163,31 +168,33 @@ func (pm *taskQueuePartitionManagerImpl) initialize(ctx context.Context) {
 			pm.config.EnableFairness = true
 		}
 	default:
-		pm.managerReady.Set(nil, serviceerror.NewInternal("Unknown FairnessState in UserData"))
+		pm.defaultQueueFuture.Set(nil, serviceerror.NewInternal("Unknown FairnessState in UserData"))
 		return
 	}
 
 	defaultQ, err := newPhysicalTaskQueueManager(pm, UnversionedQueueKey(pm.partition))
 	if err != nil {
-		pm.managerReady.Set(nil, err)
+		pm.defaultQueueFuture.Set(nil, err)
 		return
 	}
 	defaultQ.Start()
-	pm.managerReady.Set(defaultQ, nil)
+	pm.defaultQueueFuture.Set(defaultQ, nil)
 }
 
-func (pm *taskQueuePartitionManagerImpl) MustGetDefaultQueue(ctx context.Context) physicalTaskQueueManager {
-	queue, err := pm.managerReady.Get(ctx)
+func (pm *taskQueuePartitionManagerImpl) defaultQueue() physicalTaskQueueManager {
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
```

Comment:
I kinda hate that we have to this (I did it somewhere else too). this is our future implementation, I think we should just add a GetIfReady to it.